parastell/invessel_build.py

```diff
@@ -226,39 +217,71 @@ def calculate_loci(self):
 
         [surface.calculate_loci() for surface in self.Surfaces.values()]
 
+    def _check_edge(self, edge):
+        """Checks a CadQuery edge to determine if the its center of mass is
+        located at either end of the stellarator build.
+        (Internal function not intended to be called externally)
+
+        Arguments:
+            edge (object): CadQuery.Edge object.
+
+        Returns:
+            (bool): flag to indicate whether edge is located at either end of
+                stellarator build.
+        """
+        x = edge.Center().x
+        y = edge.Center().y
+
+        toroidal_angle = np.arctan2(y, x)
+        """if self.radial_build.toroidal_angles[0] >= 0.0:
+            toroidal_angle = (toroidal_angle + 2 * np.pi) % (2 * np.pi)"""
+        toroidal_angle = np.rad2deg(toroidal_angle)
+
+        toroidal_extent = [
+            self.radial_build.toroidal_angles[0],
+            self.radial_build.toroidal_angles[-1],
+        ]
+
+        return np.any(np.isclose(toroidal_extent, toroidal_angle))
+
     def generate_components(self):
         """Constructs a CAD solid for each component specified in the radial
-        build by cutting the interior surface solid from the outer surface
-        solid for a given component.
+        build by combining each component's outer surface with that of the
+        component interior to it.
         """
         self._logger.info(
             "Constructing CadQuery objects for in-vessel components..."
         )
 
-        interior_surface = None
-
-        segment_angles = np.linspace(
-            self.radial_build.toroidal_angles[-1],
-            self._repeat * self.radial_build.toroidal_angles[-1],
-            num=self._repeat,
-        )
+        inner_surface = None
 
         for name, surface in self.Surfaces.items():
             outer_surface = surface.generate_surface()
+            edge_list = [
+                edge
+                for edge in outer_surface.Edges()
+                if self._check_edge(edge)
+            ]
+            wire_list = [cq.Wire.assembleEdges([edge]) for edge in edge_list]
```

Comment:
Can't we just combine these? I don't see `edge_list` used elsewhere:
```suggestion
            wire_list = [cq.Wire.assembleEdges([edge]) for edge in outer_surface.Edges()
                if self._check_edge(edge)]
```